jetty-websocket/javax-websocket-client-impl/src/main/java/org/eclipse/jetty/websocket/jsr356/endpoints/JsrAnnotatedEventDriver.java

```diff
@@ -343,9 +338,10 @@ public void onTextFrame(ByteBuffer buffer, boolean fin) throws IOException
                     catch (Throwable e)
                     {
                         session.close(e);
+                        return;
```

Comment:
This return seem out of place. shouldn't we at least close/fail the MessageReader too?